### ISSUE TITLE
[TypeScript] "public constructor" fails to transpile in 7.7.x

### ISSUE THREAD

**Issue by lirbank** [babel/babel#10661]
## Bug Report

**Current Behavior**
After upgrading Babel to 7.7.x, the code constructors marked as "public" does not transpile.  to. If I remove the "public" keyword it works. This only occurs with constructors, public methods still transpile properly.

Error:
```
  public constructor() {
         ^^^^^^^^^^^

SyntaxError: Unexpected identifier
```

**Input Code**
- REPL or Repo link if applicable:

```js
class SomeClass {
  public constructor() {
    // Stuff
  }
}

```

**Expected behavior/code**
Using "public constructor" worked before so this seems to be a regression.

**Babel Configuration (.babelrc, package.json, cli command)**

```js
{
  "presets": [
    "@babel/typescript",
    [
      "@babel/preset-env",
      {
        "targets": {
          "node": "current"
        }
      }
    ]
  ],
  "plugins": ["@babel/proposal-class-properties"]
}
```

**Environment**
```
-    "@babel/core": "7.6.4",
-    "@babel/node": "7.6.3",
-    "@babel/preset-env": "7.6.3",
-    "@babel/preset-typescript": "7.6.0",
+    "@babel/core": "7.7.0",
+    "@babel/node": "7.7.0",
+    "@babel/plugin-proposal-class-properties": "7.7.0",
+    "@babel/preset-env": "7.7.1",
+    "@babel/preset-typescript": "7.7.0",
```

**Additional context/Screenshots**
https://www.typescriptlang.org/docs/handbook/classes.html#public-private-and-protected-modifiers

**Comment by JLHwung** [comment]
Closing it as duplicate of #10644, which will be fixed in 7.7.2. 

@lirbank Nice report. Thank you!

**Comment by ycjcl868** [comment]
+1
